src/main/kotlin/com/openlattice/data/storage/PostgresDataQueries.kt

```diff
@@ -58,51 +65,44 @@ fun buildPreparableFiltersSql(
         metadataOptions: Set<MetadataOption>,
         linking: Boolean,
         idsPresent: Boolean,
-        partitionsPresent: Boolean
+        partitionsPresent: Boolean,
+        detailed: Boolean = false
 ): Pair<String, Set<SqlBinder>> {
     val filtersClauses = buildPreparableFiltersClause(startIndex, propertyTypes, propertyTypeFilters)
     val filtersClause = if (filtersClauses.first.isNotEmpty()) " AND ${filtersClauses.first} " else ""
-
     val metadataOptionColumns = metadataOptions.associateWith(::mapMetaDataToColumnSql)
-    val nonAggregatedMetadataSql = metadataOptionColumns
-            .filter { !isMetaDataAggregated(it.key) }.values.joinToString("")
-    val innerGroupBy = groupBy(ESID_EKID_PART_PTID + nonAggregatedMetadataSql)
-    // TODO: remove IS NOT NULL post-migration
-    val linkingClause = if (linking) {
-        " AND ${ORIGIN_ID.name} IS NOT NULL AND ${ORIGIN_ID.name} != '${IdConstants.EMPTY_ORIGIN_ID.id}' "
-    } else {
-        ""
-    }
+    val metadataOptionColumnsSql = metadataOptionColumns.values.joinToString("")
+    val innerGroupBy = groupBy(ESID_EKID_PART_PTID)
```

Comment:
If ORIGIN_ID metadata option is true, it should be either in this group by or aggregated in the inner query

Reply:
ORIGIN_ID makes no sense outside of linking and shouldn't really be a metadata option. Will probably switch this if statement to be based on linking.

Long term we need to fix linking indexing to handle the new read format and not read data in the current format (which I'm not changing in this PR).

Reply:
We're missing tons of validation here. This is extending a hack that will have to be removed.

Reply:
For example, ORIGIN_IDS should not be specifiable metadata independent of linking, and it makes no sense that ORIGIN_IDS are separate concept from the detailed view of reading an object.